policy/file/source.go

```diff
@@ -0,0 +1,288 @@
+package file
+
+import (
+	"context"
+	"crypto/md5"
+	"errors"
+	"fmt"
+	"sync"
+
+	hclog "github.com/hashicorp/go-hclog"
+	multierror "github.com/hashicorp/go-multierror"
+	fileHelper "github.com/hashicorp/nomad-autoscaler/helper/file"
+	"github.com/hashicorp/nomad-autoscaler/helper/uuid"
+	"github.com/hashicorp/nomad-autoscaler/policy"
+)
+
+// Ensure NomadSource satisfies the Source interface.
+var _ policy.Source = (*Source)(nil)
+
+// Source is the File implementation of the policy.Source interface.
+type Source struct {
+	config *policy.ConfigDefaults
+	dir    string
+	log    hclog.Logger
+
+	// reloadChan is the channel that the agent sends to in order to trigger a
+	// reload of all file policies.
+	//
+	// TODO(jrasell) the reloadChan is not yet triggered by anything and needs
+	//  to be hooked up via the agent.
+	reloadChan chan bool
+
+	// policyReloadChan is used internally to trigger a reload of individual
+	// policies from disk.
+	policyReloadChan chan bool
+
+	// idMap stores a mapping between between the md5sum of the file path and
+	// the associated policyID. This allows us to keep a consistent PolicyID in
+	// the event of policy changes.
+	idMap     map[[16]byte]policy.PolicyID
```

Comment:
`[16]byte` could be a type so it's more clear what it represents (the hashed file path right?)